kong/example_test.go

```diff
@@ -0,0 +1,52 @@
+package kong_test
+
+import (
+	"fmt"
+	"os"
+
+	jxkong "foxygo.at/jsonnext/kong"
+	"github.com/alecthomas/kong"
+	jsonnet "github.com/google/go-jsonnet"
+)
+
+func Example() {
+	cfg := jxkong.NewConfig() // foxygo.at/jsonnext/kong imported as jxkong
+
+	// Define kong CLI struct embedding jxkong.Config, adding your own
+	// application-specific flags and args.
+	cli := struct {
+		jxkong.Config
+		Verbose  bool
+		Filename string `arg:""`
+	}{
+		Config: *cfg,
+	}
+
+	// Simulate command line arguments
+	os.Args = []string{
```

Comment:
would be nice to have the same detailed test case as the flag example imo.

maybe even have a bunch of tests run with flag and kong parsing ensuring the results are identical?

Reply:
There is no flag example - i guess that's a shortcoming I should address. The tests for flags covers every flag - I wonder why I didn't do it with kong? I should do that I guess.

I think using every flag in an example is too much, but I think I should add the same comprehensive tests for kong. Will do.

Reply:
Quoting you from Slack for poterity:

>  For the kong tests, I am still thinking about how to do that. I think I will write a test suite package that can be used by the individual tests so the same tests can be run against any cli parsing package. This means if another were to be written for, say kingpin or spf13/cobra, etc, it should be a matter of plugging in the setup into the test suite and running it, and it will validate that the CLI was parsed as expected. 

imo that's a really good idea!